A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDInfo.js

```diff
@@ -0,0 +1,531 @@
+          this.currentWindAngle = smoothedAngle % 360;
+          if (this.windDirection != null) {
+              if (windStrongEnough)
+                  this.windDirection.textContent = this.currentWindAngle.toFixed(0).padStart(3, "0");
+              else
+                  this.windDirection.textContent = "---";
+          }
+      }
+      if (refreshWindStrength) {
+          this.currentWindStrength = _windStrength;
+          if (this.windStrength != null) {
+              this.windStrength.textContent = this.currentWindStrength.toString().padStart(2, "0");
+          }
+      }
+      if (refreshWindArrow) {
+          this.currentPlaneAngle = _planeAngle;
+          if (this.windArrow != null) {
+              {
+                  var arrowAngle = this.currentWindAngle - this.currentPlaneAngle;
+                  arrowAngle += 180;
+                  var transformStr = this.windArrow.getAttribute("transform");
+                  var split = transformStr.split("rotate");
+                  if ((split != null) && (split.length > 0)) {
+                      this.windArrow.setAttribute("transform", split[0] + " rotate(" + arrowAngle + ")");
+                  }
+                  this.windArrow.style.display = "block";
+              }
+          }
+      }
+  }
+  setWaypoint(_name, _track, _distance, _eta, _force = false) {
+      if (this.waypoint) {
+          if (this._navMode == Jet_NDCompass_Navigation.NAV) {
+              if (_name && _name != "") {
+                  if (this.waypointName != null) {
+                      this.waypointName.textContent = _name;
+                  }
+                  if ((_track != this.currentWaypointTrack) || _force) {
+                      this.currentWaypointTrack = _track;
+                      if (this.waypointTrack) {
+                          this.waypointTrack.textContent = this.currentWaypointTrack.toString().padStart(3, "0") + String.fromCharCode(176);
+                      }
+                  }
+                  if ((_distance != this.currentWaypointDistance) || _force) {
+                      this.currentWaypointDistance = _distance;
+                      if (this.waypointDistance != null) {
+                          if (this.currentWaypointDistance < 10000)
+                              this.waypointDistance.textContent = this.currentWaypointDistance.toFixed(1);
+                          else
+                              this.waypointDistance.textContent = this.currentWaypointDistance.toFixed(0);
+                      }
+                  }
+                  if ((_eta != this.currentWaypointTimeETA) || _force) {
+                      this.currentWaypointTimeETA = _eta;
+                      const utcETA = (_eta + SimVar.GetGlobalVarValue("TIME ZONE OFFSET", "seconds")) % 86400;
+                      if (this.waypointTime != null) {
+                          const hours = Math.floor(utcETA / 3600).toString().padStart(2, "0");
```

Comment:
Please consider keeping the implementation as it was; first calculate the value (original code as a reference)
```
var hours = Math.floor(localETA / 3600);
```
as that value is not tied to the UI representation. Only then create the presentation format when needed
```
this.waypointTime.textContent = hours.toString().padStart(2, "0")...
```
This way the actual value is re-usable and it's clear the what line does what. Now value calculation is mixed up with the presentation.

Just my 2 cents.

Reply:
yeahh thats a fair point. reverted.